Framework/Core/include/Framework/ASoA.h

```diff
@@ -1664,7 +1668,9 @@ constexpr auto is_binding_compatible_v()
     template <typename T>                                                                                                         \
     auto _Getter_##_as() const                                                                                                    \
     {                                                                                                                             \
-      assert(mBinding != nullptr);                                                                                                \
+      if (O2_BUILTIN_UNLIKELY(mBinding == nullptr)) {                                                                             \
+        throw o2::framework::runtime_error("Index to " #_Table_ " is not bound! Did you subscribe to the table?");                \
+      }                                                                                                                           \
       if (O2_BUILTIN_UNLIKELY(!has_##_Getter_())) {                                                                               \
```

Comment:
We should probably hide all this is a separate, non templated, function.

Reply:
@ktf something like this?